Feature/test imputer

philipp-jung opened the pull request:
I add two basic integration tests. They are inspired by the old `simple_imputer`'s tests. Both tests currently fail:
- `test_quantile_regression` is testing the experimental quantile regression feature. The way the `assert()` clause is set up doesn't really make any sense, so this is expected to fail.
- `test_precision_threshold` is failing, too. This is due to the fact that the imputer returns a `nan` at a user-specied precision-threshold of 0.01. Making me believe that autowig isn't working the way it's supposed to just yet.

philipp-jung committed:
Initial cleanup of the complete method

philipp-jung committed:
[wip] test thresholds

philipp-jung committed:
[wip] start implementing test

We're experiencing some unexpected behavior regarding the way we
determine thresholds in `.fit()`. Which is why I've started looking into
adding some tests to the code.

The first test is a very basic integration test. It is supposed to
ensure that the core feature of Autowig, thresholding and imputing
values with that, works.

I got inspired by the integration tests written for the old
`SimpleImputer`. The current state of the branch doesn't properly work.

philipp-jung committed:
Add two integration tests

There are now two integration tests. Both currently fail, because
`imputer.predict()` returns missing values. Missing values are returned
because a class probability is below the computed precision threshold.

- `test_precision_threshold` tests the ability to impute categorical
data. The precision threshold is set to `0.01`, which makes me believe
that this hints at a bug.

- `test_quantile_regression` tests the ability to impute numerical
data. The precision threshold is set to `0.01`, which also makes me
believe that this hints at a bug.

felixbiessmann reviewed (changes requested):
I would put the tests in a separate test directory and would probably not include the model files in the repo - for testing the deserialization one could just generate a serialized model.

philipp-jung committed:
Merge Felix' code into feature/test-imputer

philipp-jung committed:
[wip] merge branches, clean & improve

I merged the two branches. Since Felix changed the API a bunch, I ended
up changing the code a bit while doing this. The `complete()` method is
defunctional at the moment.

I cleaned up the notebook a bit, moved the tests I've written to the
tests directory. The tests still don't work, this needs some
addiontional work.

I now catch when users call `predict()` before calling `fit`.

philipp-jung committed:
Refine Tests

There are now two tests in place: An integration test for quantile
regression, and a test for precision thresholds in classification tasks.

The test for quantile regression is just a scaffolding for now, as the
API is still a bit weird and I expect it to change in the future -- we
compute the quantile-regresssion thresholds during `.fit()`, but the
classification thresholds during `.predict()`. I find that odd.

I added an assertion to `test_autogluon_imputer_precision_threshold`
that ensures that empirically measured precision thresholds increase
monotonously with increased user-specified precision thresholds.

felixbiessmann reviewed (approved):
looks great
